include/qfactory.hpp

```diff
@@ -24,6 +25,28 @@
 namespace Qrack {
 
 /** Factory method to create specific engine implementations. */
+template <typename... Ts>
+QInterfacePtr CreateQuantumInterface(
```

Comment:
I'm about 85% sure that you don't need these variations - dealing with this is exactly why the variadic template parameters were used.

Reply:
However, one or more engine type parameters are completely dropped from the non-variadic parameters, depending on how many engine type arguments are available in constructors, but the template requires that normalized constructors be available for the entire set of variadic parameters. 60% sure it won't work without superfluous constructors, but I'll try.

Reply:
This requires normalized constructors to work. If one object's constructor fits the variadic template, every object's constructor in the method has to fit the same _case_ _in_ _usage_, whether or not the switch will every actually construct that object, in practice.